src/chocolatey/infrastructure/commands/IListCommand.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright © 2015 - Present RealDimensions Software, LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// 
+// You may obtain a copy of the License at
+// 
+// 	http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace chocolatey.infrastructure.commands
+{
+    using System.Collections.Generic;
+    using app.configuration;
+
+    public interface IListCommand<out T> : ICommand
+    {
+        IEnumerable<T> list(ChocolateyConfiguration config, bool logResults = true);
```

Comment:
> You can't add an overload of a method where the only difference is the return value, so you'd have to redefine run -- either way, if you put it on the ICommand instead of on a derived interface, then you have to implement it on every command even if there is no benefit.

So basically we are saying redefine run. I think by the time we are done, most of them will likely return something. Perhaps keep it separate for now though.